From 57f53e754af180f799a017e34d45c79d9830b88b Mon Sep 17 00:00:00 2001
From: Matthieu Vanin <matthieu.vanin@nxp.com>
Date: Wed, 26 Sep 2018 09:29:22 -0400
Subject: [PATCH 1/2] HDCP - Disable HDCP support from the sink

Signed-off-by: Matthieu Vanin <matthieu.vanin@nxp.com>
---
 wfd.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wfd.sh b/wfd.sh
index d1065d6..fb90a5d 100755
--- a/wfd.sh
+++ b/wfd.sh
@@ -52,7 +52,7 @@ start_function() {
         exit 0
     fi
 
-    export launch="wfd_sink -i $IFACE -w $WPA_SUPPLICANT_CONTROL_IFACE -n $SSID -p $CHN -m $BEHAVE -e -u"
+    export launch="wfd_sink -i $IFACE -w $WPA_SUPPLICANT_CONTROL_IFACE -n $SSID -p $CHN -m $BEHAVE -u"
     echo "$launch"
     $launch &
 
-- 
1.9.1

