From e1713a397e9e3099030292a6c33a24f352928bb4 Mon Sep 17 00:00:00 2001
From: Matthieu Vanin <matthieu.vanin@nxp.com>
Date: Wed, 31 May 2017 14:34:57 -0400
Subject: [PATCH] HDCP - Disable HDCP support from the sink

---
 wfd.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/etc/init.d/wfd.sh b/etc/init.d/wfd.sh
index fe78e65..aa57a21 100755
--- a/wfd.sh
+++ b/wfd.sh
@@ -64,7 +64,7 @@ start_function() {
     fi
 
     # wfd_sink -i wlan0 -n wfd-nxp &
-    export launch="wfd_sink -i $IFACE -w $WPA_SUPPLICANT_CONTROL_IFACE -n $SSID -p $CHN -m $BEHAVE -e -u"
+    export launch="wfd_sink -i $IFACE -w $WPA_SUPPLICANT_CONTROL_IFACE -n $SSID -p $CHN -m $BEHAVE -u"
     echo "$launch"
     $launch &
 
-- 
1.9.1

