From def2b2d271a22086084d099ae0996ea8e87d8c84 Mon Sep 17 00:00:00 2001
From: Matthieu Vanin <matthieu.vanin@nxp.com>
Date: Thu, 20 Sep 2018 20:19:35 -0400
Subject: [PATCH 1/2] wfd_manager: enable automatic screen connection by
 default

Signed-off-by: Matthieu Vanin <matthieu.vanin@nxp.com>
---
 wfd_manager.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wfd_manager.sh b/wfd_manager.sh
index 906daa6..698f503 100755
--- a/wfd_manager.sh
+++ b/wfd_manager.sh
@@ -31,7 +31,7 @@ start_function() {
     #wfd_manager --dfb:fbdev=/dev/fb0 &
     #wfd_manager -s freescale &
     #WAYLAND_DISPLAY=wayland-0 XDG_RUNTIME_DIR=/run/user/root wfd_manager_qt -platform eglfs &
-    WAYLAND_DISPLAY=wayland-0 XDG_RUNTIME_DIR=/run/user/root wfd_manager_qt -platform wayland &
+    WAYLAND_DISPLAY=wayland-0 XDG_RUNTIME_DIR=/run/user/root wfd_manager_qt -a -platform wayland &
     pid=$!
 
     if [ "$pid" != "" ]; then
-- 
1.9.1

